### PR TITLE
Enhance exported warnings to include i18n token

### PR DESCRIPTION
**Pull request by Mathieuka**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -85,14 +85,41 @@ export async function runASTAnalysisOnFile(pathToFile, options = {}) {
...
-    unsafeRegex: "unsafe-regex",
-    unsafeStmt: "unsafe-stmt",
-    unsafeAssign: "unsafe-assign",
-    encodedLiteral: "encoded-literal",
-    shortIdentifiers: "short-identifiers",
-    suspiciousLiteral: "suspicious-literal",
-    obfuscatedCode: "obfuscated-code"
+    parsingError: {
+      code: "ast-error",
+      i18n: "ast_error"
```
Comment (fraxken) [inline index.js]: `warnings.ast_error`. You need the full i18n path here (same for all the i18n properties below).

**Review by fraxken — CHANGES_REQUESTED** [review]
You also need to update the TypeScript definition file:
https://github.com/NodeSecure/js-x-ray/blob/f833d69b3f692a2c87ab3c48067a425dae76e14d/index.d.ts#L57-L67

Maybe we can also add a section in README to explain how we can use CONSTANTS warnings with the i18n code ?

File: README.md [hunk 2]
```diff
@@ -73,6 +73,44 @@ The analysis will return: `http` (in try), `crypto`, `util` and `fs`.
...
+We make a call to `i18n` through the package `NodeSecure/i18n` to get the translation.
+```
+import * as scanner from "@nodesecure/scanner";
+import * as i18n from "@nodesecure/i18n";
+
+const payload = await scanner.from("packageName", { 4 }, initLogger("packageName"));
+
+if (payload.warnings.length) {
+  const value = i18n.getToken(payload.warnings[0].i18n);
+}
```
Comment (fraxken) [inline README.md]:
```js
import * as jsxray from "@nodesecure/js-x-ray";
import * as i18n from "@nodesecure/i18n";

console.log(i18n.getToken(jsxray.CONSTANTS.Warnings.parsingError.i18n));
```

An example like this is better (payload.warnings is not about JS-X-Ray warnings).

Comment (Mathieuka) [reply README.md]: Yes

File: index.js [hunk 3]
```diff
@@ -85,14 +85,41 @@ export async function runASTAnalysisOnFile(pathToFile, options = {}) {
 
 export const CONSTANTS = {
```
Comment (Kawacrepe) [inline index.js]: Do we want to keep exporting `export const CONSTANTS = {
  Warnings: ...
}`

Instead exporting directly the warnings object

cf: our discussion on discord #js-x-ray chan

Comment (Mathieuka) [reply index.js]: Yes !

**Review by fraxken — APPROVED** [review]
LGTM

**Review by Kawacrepe — APPROVED** [review]
neat!

**Pull request merged**